### PR TITLE
Add soak tests for GenericMapStore [HZ-1515]

### PR DESCRIPTION
**Pull request by sumnerib**
Adds a new soak test to execute map operations (get/put/delete) which trigger GenericMapStore operations against a MySQL DB.

### CODE REVIEW

**Review by olukas — APPROVED** [review]
LGTM, just one minor comment. Thanks.

File: generic-map-store-test/src/main/java/com/hazelcast/jet/tests/mapstore/GenericMapStoreTest.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    private void runAndVerifyMapStoreOperations(int key) {
+        // MapStore.store()
+        GenericRecord person = map.get(key);
+        AbstractSoakTest.assertEquals("name-" + key, person.getString("name"));
+
+        // MapStore.delete()
+        map.delete(key);
+
+        // MapStore.load()
+        AbstractSoakTest.assertTrue(map.put(key, person) == null);
```
[the remark this answers is not in the document]
Comment (sumnerib) [reply generic-map-store-test/src/main/java/com/hazelcast/jet/tests/mapstore/GenericMapStoreTest.java:116]: I actually want `assertNull()`. I can add it to `AbstractSoakTest`. Thanks!

File: generic-map-store-test/pom.xml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            <artifactId>hazelcast-mapstore</artifactId>
+            <version>${hazelcast.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-reload4j</artifactId>
+            <version>1.7.36</version>
+        </dependency>
+    </dependencies>
+</project>
```
Comment (Patras3) [inline generic-map-store-test/pom.xml:104]: nit: No newline at end of file

**Review by Patras3 — APPROVED** [review]
LGTM, Just one nit (missing empty line at the end of file.)

**Pull request merged**